### PR TITLE
channel_whitelist: channels Stampy can respond to

### PR DESCRIPTION
**Pull request by ProducerMatt**
Depends on #301

### CODE REVIEW

File: servicemodules/discord.py [hunk 1]
```diff
@@ -147,6 +148,14 @@ async def on_message(
                 )
                 self.utils.last_message_was_youtube_question = False
 
+            #log.info("Checking whitelist...") # DEBUG
+            if (isinstance(channel_whitelist, frozenset)
+                    and hasattr(message.channel, "id")
+                    and str(message.channel.id) not in channel_whitelist):
+                #log.info("message channel {} not in whitelist".format(message.channel.id)) # debug
+                return None
```
Comment (Aprillion) [inline servicemodules/discord.py:156]: direct message will continue to work?

Comment (ProducerMatt) [reply servicemodules/discord.py:156]: Oh shoot good catch. Thanks

**Comment by ProducerMatt** [comment]
Uhh why did it close

**Comment by ProducerMatt** [comment]
Reopen pls

**Comment by ProducerMatt** [comment]
Oh duh. It's trying to merge into the talkative branch

**Pull request closed without merging**